File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Membership/MembershipFeature.cs
namespace Psi.Data.Models.ClientConfigurationModels.Membership
{
    public class MembershipFeature : SettingsBaseHelper
    {
        private MemberProfile _memberProfile;
        private Flags _flags;

        public MembershipFeature(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Membership.MembershipEnabled")]
        public bool MembershipEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Membership.MinVersion")]
        public double MinVersion
        {
            get { return GetDoubleValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Membership.GetMyCUClubSettingsFromDatabase")]
        public bool GetMyCUClubSettingsFromDatabase
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        public MemberProfile MemberProfile
        {
            get => _memberProfile ?? (_memberProfile = new MemberProfile(SettingsBase));
            set => _memberProfile = value;
        }

        public Flags Flags
        {
            get => _flags ?? (_flags = new Flags(SettingsBase));
            set => _flags = value;
        }

        [SettingKey("X.App.HomeBanking.AllowForeignAddressUpdate")]
        public bool ShouldAllowForeignAddressUpdate
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Membership/MembershipFeature.ts
// Generated imports
import { MemberProfile } from './MemberProfile';
import { Flags } from './Flags';

export interface MembershipFeature {
    /** @settingKey Membership.MembershipEnabled */
    membershipEnabled: boolean;
    /** @settingKey Membership.MinVersion */
    minVersion: number;
    /** @settingKey Membership.GetMyCUClubSettingsFromDatabase */
    getMyCUClubSettingsFromDatabase: boolean;
    memberProfile: MemberProfile;
    flags: Flags;
    /** @settingKey X.App.HomeBanking.AllowForeignAddressUpdate */
    shouldAllowForeignAddressUpdate: boolean;
}
